src/dbally/collection.py

```diff
@@ -238,6 +239,10 @@ def get_similarity_indexes(self) -> Dict[AbstractSimilarityIndex, List[Tuple[str
         indexes: Dict[AbstractSimilarityIndex, List[Tuple[str, str, str]]] = {}
         for view_name in self._views:
             view = self.get(view_name)
+
+            if not issubclass(view.__class__, BaseStructuredView):
```

Comment:
`if not isinstance(view, BaseStructuredView):` would seems like more direct solution.